dart/biomechanics/DynamicsFitter.cpp

```diff
@@ -12322,6 +12330,19 @@ bool DynamicsFitter::timeSyncAndInitializePipeline(
   }
   multimassZeroLinearResidualsOnCOMTrajectory(init, maxTrialsToSolveMassOver);
 
+  if (trimMissingGRFs) {
+    std::cout << "Triming timestamps with missing GRFs..." << std::endl;
+    std::vector<std::pair<int,int>> newIndicesTrials =
+        init->trimMissingGRFsAtEndpoints();
+    // Trim the "original" pose trials data to match the new trial lengths.
+    for (int i = 0; i < init->poseTrials.size(); i++){
+      std::pair<int,int> newIndices = newIndicesTrials[i];
+      originalPoseTrials[i] = DynamicsInitialization::trimColumns(
```

Comment:
Ok, I strongly suspect this is part of the problem. We want to be updating `init` with the newly trimmed state, right? In the `init` object we've got shitloads of time-series data that would all need to be updated in sync, like poses, GRFs, markers, IMUs, etc etc. I haven't finished reading, but just jotting down my thoughts.

Reply:
Oh, I misread. So you trim above with `trimMissingGRFsAtEndpoints()`. Ignore my above comment.